core/src/OpenMPTarget/Kokkos_OpenMPTarget_Macros.hpp

```diff
@@ -23,6 +23,14 @@
 #if defined(KOKKOS_COMPILER_CLANG) && (KOKKOS_COMPILER_CLANG >= 1700) && \
     (defined(KOKKOS_ARCH_AMD_GPU) || defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU))
 #define KOKKOS_IMPL_OPENMPTARGET_LLVM_EXTENSIONS
+
+// Define a macro that can be used to separate Kernel Mode extensions in llvm
+// compiler from OpenMP standard directives. The extensions are only available
+// from llvm compiler version greater than version 20.
+#if (KOKKOS_COMPILER_CLANG >= 2000)
+#define KOKKOS_IMPL_OPENMPTARGET_KERNEL_MODE
```

Comment:
Pick your poison.  Either this is defined in the configuration file or your deduce it in that file but not both

Reply:
So thats one thing I am not sure about. If I deduce it then a user cannot avoid their use unless they change the source code. But I also cannot think of a reason why they would want to avoid using better implementation except for some research. Ok, I will deduce it myself.